org.opent2t.onboarding.getverahub/js/thingOnboarding.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+var https = require('https');
+var cryptojs_sha1 = require('crypto-js');
+var request = require('request-promise');
+
+class Onboarding {
+
+    ///
+    /// gets the session token for the provided server url
+    ///
+    getSessionToken(url) {
+        return this.makeVeraRequest(url, this.veraSessionTokenPath, false, false, this.mmsInfo, true);
+    }
+
+    ///
+    /// gets the mms info for the user account
+    ///
+    getMmsAuthInfo() {
+        var path = '/autha/auth/username/' + this.username + '?SHA1Password=' + this.sha1Password + '&PK_Oem=1';
+        
+        return this.makeVeraRequest(this.veraAccountUrl, path)
+            .then((data) => {
+                this.mmsInfo = data;
+                return data;
+            });
+    }
+
+    ///
+    /// gets the device specific info, including the relay server with which to talk to devices through
+    ///
+    getDeviceInfo(url, sessionToken, pkDevice) {
+        var path = '/device/device/device/' + pkDevice;
+        return this.makeVeraRequest(url, path, sessionToken);
+    }
+
+    makeVeraRequest(url, path, sessionToken, keepAlive, mmsInfo, returnRawBody) {
+        console.log("[VeraHub] -------------------------")
+        console.log("[VeraHub] makeVeraRequest: " + url)
+        console.log("[VeraHub] path           : " + path);
+        console.log("[VeraHub] sessionToken   : " + sessionToken);
+        console.log("[VeraHub] keepAlive      : " + keepAlive);
+        console.log("[VeraHub] mmsInfo        : " + mmsInfo);
+        console.log("[VeraHub] returnRawBody  : " + returnRawBody);
+
+        var completeUrl = "https://" + url + path;
+
+        var options = {
+            url: completeUrl,
+            method: 'GET'
+        };
+
+        if (sessionToken) {
+            options.headers = { 'MMSSession': sessionToken };
+        }
+        else if (mmsInfo) {
+            options.headers = {
+                'MMSAuth': mmsInfo.Identity,
+                'MMSAuthSig': mmsInfo.IdentitySignature
+            };
+        }
+
+        if (keepAlive) {
+            var keepAliveAgent = new https.Agent({keepAlive: true});
+            options.agent = keepAliveAgent;
+        }
+
+        return request(options)
+            .then(function (body) {
+                if (returnRawBody) {
+                    return body;
+                }
+                else {
+                    return JSON.parse(body);
+                }
+            });
+    }
```

Comment:
Correct it will -- and just to be sure i'll work on adding some unit tests for some of our translators

---
In reply to: [103803949](https://github.com/openT2T/onboarding/pull/32#discussion_r103803949) [](ancestors = 103803949,103618448)